tests/unit/distributed_test.cc

```diff
@@ -142,7 +142,7 @@ SEASTAR_TEST_CASE(test_map_reduce_lifetime) {
         }
         auto operator()(const X& x) {
             return yield().then([this, &x] {
-                BOOST_REQUIRE(!destroyed);
+                assert(!destroyed);
```

Comment:
> BOOST_REQUIRE macros are not thread safe, so replace them with
asserts in multithreaded environment

could you share the impact of "not thread safe" ?

Reply:
1. ThreadSanitizer warns
2. [Boost.Test docs](https://live.boost.org/doc/libs/master/libs/test/doc/html/boost_test/section_faq.html) say about it

Reply:
Normal Seastar tests run on a single reactor thread. But this specific test deliberately starts the same code on multiple threads. And then, supposedly, you're not supposed to use BOOST_REQUIRE from multiple threads.

However, I don't think replacing BOOST_REQUIRE by assert() is the right thing to do - it will replace a clean test failure of a single test function, by a crash of the entire executable... :-(

I think it more sense to have an (atomic-variable) counter of errors, and then BOOST_REQUIRE that the counter be 0 when the do_with_distributed ends (on a single thread).

Reply:
BTW, I find it really shocking - that in 2024 Boost.Test doesn't support multiple threads out of the box. It's trivial to support (especially when performance is not even an issue). It reduces whatever little confidence I had in this test framework :-(